tests/file_storage/tests.py

```diff
@@ -866,6 +872,52 @@ def test_stringio(self):
             self.assertEqual(f.read(), b'content')
 
 
+class FieldCallableFileStorageTests(SimpleTestCase):
+    def setUp(self):
+        self.temp_storage_location = tempfile.mkdtemp(suffix='filefield_callable_storage')
+
+    def tearDown(self):
+        shutil.rmtree(self.temp_storage_location)
+
+    def test_callable_base_class_error_raises(self):
+        class NotStorage:
+            pass
+        msg = "FileField.storage must be a subclass/instance of django.core.files.storage.Storage"
+        for type_ in (NotStorage, str, list, set, tuple):
+            with self.subTest(type_=type_):
+                with self.assertRaisesMessage(ImproperlyConfigured, msg):
+                    FileField(storage=type_)
+
+        # These should not raise, and should use default_storage
+        for type_ in (None, False):
+            with self.subTest(type_=type_):
+                obj = FileField(storage=type_)
+                self.assertEqual(obj.storage, default_storage)
```

Comment:
These assertions are not related with this patch. Personally I don't see much value in adding them.

Reply:
It does cover a previously untested scenario though, afaik. This assures that this and any future patch does not break this behavior retroactively.

Reply:
`False` is not an expected value, it works as a side effect of the current implementation. We shouldn't add tests for passing `False` because it can create an impression that this is expected and supported.

About `None`, I would rather add test for
```python
self.assertEqual(FileField().storage, default_storage)
```
Nevertheless it's not related with this new feature and should be moved to a separate commit.